code/modules/religion/darkness_sect.dm

```diff
@@ -121,13 +126,16 @@
 	. = ..()
 	STOP_PROCESSING(SSobj, src)
 
-/datum/component/dark_favor/process()
-	if(!istype(parent, /atom) || !istype(creator))
+/datum/component/dark_favor/process(delta_time)
+	var/datum/religion_sect/darkness_sect/sect = GLOB.religious_sect
+	if(!istype(parent, /atom) || !istype(creator) || !istype(sect))
```

Comment:
you should check of types when initializing the component and return failure if not